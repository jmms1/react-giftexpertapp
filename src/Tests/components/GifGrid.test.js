import React from 'react';
import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')
import '@testing-library/jest-dom'


describe('Prueba del componente <GifGrid />', () => {
    
    const value = 'Hola Mundo';
    
    
    test('Compomente se debe renderizar correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true 
        })
        const wrapper = shallow( <GifGrid category={value} />)
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('Debe mostrar items cuando carga imagenes useFetchGifs', () => {
        
        const gifs = [
            {
            id: 'abv',
            title: 'abc',
            url: 'http://localhost/cualquier/cosa.jpg' 
            },
            {
                id: 'xyz',
                title: 'xyz',
                url: 'http://localhost/cualquier/cosa.jpg' 
                },
        ]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false 
        })
        const wrapper = shallow( <GifGrid category={value} />);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').lenght).toBe( gifs.lenght);

        
    })
    



})
