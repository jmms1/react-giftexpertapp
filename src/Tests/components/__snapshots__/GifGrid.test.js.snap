// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifGrid /> Compomente se debe renderizar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInY"
  >
    Hola Mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba del componente <GifGrid /> Debe mostrar items cuando carga imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInY"
  >
    Hola Mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abv"
      key="abv"
      title="abc"
      url="http://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="xyz"
      key="xyz"
      title="xyz"
      url="http://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
